src/niri.rs

```diff
@@ -3576,6 +3573,39 @@ impl Niri {
         }
     }
 
+    #[cfg(feature = "xdp-gnome-screencast")]
+    fn send_frame_callbacks_for_cast(&self, output: &Output) {
+        use smithay::wayland::seat::WaylandFocus;
+
+        let _span = tracy_client::span!("Niri::send_frame_callbacks_for_cast");
+        let frame_callback_time = get_monotonic_time();
+
+        for mapped in self.layout.windows_for_output(output) {
+            for cast in &self.casts {
+                if !cast.is_active.get() {
+                    continue;
+                }
+                let CastTarget::Window { id } = cast.target else {
+                    continue;
+                };
+                if mapped.id().get() != id {
+                    continue;
+                };
+                let Some(surface) = mapped.window.wl_surface() else {
+                    continue;
+                };
+
+                send_frames_surface_tree(
+                    surface.as_ref(),
+                    output,
+                    frame_callback_time,
+                    FRAME_CALLBACK_THROTTLE,
+                    |_, _| Some(output.clone()),
```

Comment:
I think instead of being a separate function, this cast check should be integrated into `send_frame_callbacks()` where it should be part of the `if current_primary_output.as_ref() != Some(output)` check. The stuff in `send_frame_callbacks()` does throttling to at most one frame callback per output refresh; without it windows will busy-loop frame callbacks (especially so off-screen windows, which is the primary case here).